File: frontend/src/data/EntryDto.ts
import {CommentDto} from '@/data/CommentDto';

export interface EntryDto {
    id: number;
    content: string;
    author: string;
    hashTags: string[];
    upvotes: number;
    timeStamp: Date;
    embedContentType: string;
    embedContent: string;
    comments: CommentDto[];
}

File: src/main/java/dmcs/summer/entry/EntryDto.java
package dmcs.summer.entry;

import dmcs.summer.user.User;
import lombok.Data;

import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import java.util.Date;
import java.util.List;
import java.util.Set;

@Data
public class EntryDto {
    private long id;
    private String content;
    private String author;
    private Set<String> hashTags;
    private int upvotes;

    private Date timeStamp;
    private String embedContent;

    private EMBED_CONTENT_TYPE embedContentType;
    private List<CommentDto> comments;
    private boolean canUpvote;
}
